Update design doc for PCE

rozdy opened the pull request:
Related to issue #2894

timofei-durakov commented on docs/design/path-computation.md:
```diff
@@ -28,12 +18,15 @@ The pce project has multiple unit tests that excercise most / all of the core lo
 
-In addition to this sparse guide, there are more notes in PathComputationTests.java regarding how the acceptance is executed.  That should give you a good guide to how the code comes together.
+Strategy for each path computation should be provided explicitly. If flow
+create/update/reroute request doesn't has path strategy parameter then default 
```
reroute should use flow specified param, please separate reroute from create/update

timofei-durakov replied:
to make it clear by the moment of reroute some strategy will be already stored in flow itself, so we just reuse it

rozdy replied:
Reworked doc.

rozdy commented:
@surabujin 
> 
> > ```
> > Only links with **flow encapsulation type** support and sufficient bandwidth are used as available network resources.
> > ```
> 
> **flow encapsulation type** support => appropriate **flow encapsulation type**
> are => can be
> 
> 
> > ```
> > Manually assigned value for each link. If link is under maintenance and/or link is unstable then it's cost is increased by configurable weights.
> > ```
> 
> ... by preconfigured value.
> 

Fixed.

> _[docs/design/pce/design.md, line 55 at r1](https://reviewable.io/reviews/telstra/open-kilda/2895#-LsqidaR1GHEWPDIY5t8:-LsqidaR1GHEWPDIY5t9:b3an8cv) ([raw file](https://github.com/telstra/open-kilda/blob/2288554774d8ef4bc6122abd3fff36c74995d2d6/docs/design/pce/design.md#L55)):_
> 
> > ```gfm
> > * Latency
> > 
> > Automatically calculated value for each link. Configurable weight penalties is used when calculating link weight based on latency in a similar way as for cost.
> > ```
> 
> So... we are going to covert "latency"(time unit) value into "cost" or ... we are going to convert our "penalty" modifieds into "time" units and add to the measured latency value?

I guess separate modifiers for latency should be introduced. Updated doc accordingly.

timofei-durakov commented on docs/design/pce/design.md:
```diff
@@ -0,0 +1,57 @@
+    1. if node is destination
+        1. if path weight is lower then current best 
+            1. save new path as current best path
+        2. skip further processing of the current node
```
nit: this means else? I'd make it explicit here and below

rozdy replied:
No, just next step in "then" block. Added explicit keywords.

timofei-durakov commented on docs/design/pce/design.md:
```diff
@@ -0,0 +1,57 @@
+For now two strategies are available:
+* Cost
+
+Manually assigned value for each link. If link is under maintenance and/or link is unstable then it's cost is increased by preconfigured value.
```
why not to Cost = Link Cost + Maint penalty + Unstable Penalty, where Main penalty = 0 if link isn't under maintenance...

rozdy replied:
Added example with a formula.

timofei-durakov commented on docs/design/pce/design.md:
```diff
@@ -0,0 +1,57 @@
+
+* Latency
+
+Automatically calculated value for each link. Separate maintenance/unstable penalties is used when calculating link weight based on latency in a similar way as for cost.
```
this is very confusing, latency is something in ms/ns, maintenance/unstable penalties are some abstract numbers, how are you gonna sum them up?

rozdy replied:
Correct, latency is measured in some time units. But when we use it to calculate abstract weight it becomes just a number (long). And then we will sum it with some other numbers (penalties). It's a big question on how to choose correct values and I guess it's out of scope of this design.

timofei-durakov commented on docs/design/solutions/pce-weights-strategies/pce-weights-strategies.md:
```diff
@@ -1,23 +1,22 @@
+## Features
+- Introduced PCE data model instead of reusing DAO: for storing graph weighs by features and more convenient logging
+- Decoupled AvailableNetwork creation: it has own building strategies.
+- Introduced weight computing strategies: edge data model to long functions, defines what weigh features will be used in path computation and in what proportions
```
I believe this should be covered in details

rozdy replied:
Added some details. Note: it's already done features. I just reworked this doc from "todo-style" to "done-style".

rozdy commented on docs/design/pce/design.md:
```diff
@@ -0,0 +1,63 @@
+Breadth-first search with path weight calculation and path length limitation 
+is used to calculate best path. 
+
+Starting from source node we visit all neighbor nodes and saving path to all found
```
It's American English vs British English spelling :)
